Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM ubuntu:16.04 as builder
+LABEL author="Abdelrahman Hosny <abdelrahman.hosny@hotmail.com>"
+
+RUN apt-get update && apt-get install -y build-essential \
+    clang \
+    bison \
+    flex \
+    libreadline-dev \
+    gawk \
+    tcl-dev \
+    libffi-dev \
+    git \
+    graphviz \
+    xdot \
```

Comment:
Do we need graphviz/xdot when building?

Reply:
(this likely pulls in X11, which we don't want)

Reply:
removed.